### PR TITLE
standardnotes: 3.173.4 -> 3.173.20, stop using nixpkgs electron

### LINKED ISSUE #247053: standardnotes: Selected text isn't highlighted
### Describe the bug

In standardnotes 3.167.2 from nixos-unstable, when selecting text, there's no visual indicator that the text is selected. It can still be cut, copied, dragged, etc, but it isn't highlighted in any way, making it very difficult to actually do anything involving selecting text.

### Steps To Reproduce
1. Create a note in standardnotes
2. Type some text in the note
3. Select the text

### Expected behavior
The selected text should be highlighted.

### Additional context
This bug is present whether I'm running standardnotes normally or in wayland-native mode, and is present when I update the derivation to use the latest version 3.167.28. When I revert #241831 and then re-run the update script to get the latest standardnotes using the upstream AppImage, the bug disappears, so it seems to be specifically tied to nixpkgs' electron.

### Notify maintainers

@mgregoire @chuangzhu @squalus

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.33-rt11, NixOS, 23.11 (Tapir), 23.11.20230802.66aedfd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.15.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by duxovni**
## Description of changes

Reverts #241831 to fix #247053, and updates standardnotes to the latest upstream version.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/standardnotes/default.nix [hunk 1]
```diff
@@ -1,45 +1,38 @@
...
+    libsecret
+  ];
 
-    makeWrapper ${electron}/bin/electron $out/bin/standardnotes \
-      --add-flags $out/share/standardnotes/app.asar \
-      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libsecret stdenv.cc.cc.lib ]}
+  extraInstallCommands = ''
+    # directory in /nix/store so readonly
+    cd $out
+    chmod -R +w $out
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/standardnotes/default.nix:29]:
```suggestion
    chmod -R +w $out
```
we don't need to cd here

File: pkgs/applications/editors/standardnotes/default.nix [hunk 2]
```diff
@@ -1,45 +1,38 @@
...
+    mv $out/bin/${name} $out/bin/${pname}
 
+    # fixup and install desktop file
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/standardnotes/default.nix:32]:
```suggestion
```
that's obvious from the code

File: pkgs/applications/editors/standardnotes/default.nix [hunk 3]
```diff
@@ -1,45 +1,38 @@
...
+    inherit name src;
+  };
 
-  unpackPhase = "dpkg-deb --fsys-tarfile $src | tar -x --no-same-permissions --no-same-owner";
+  nativeBuildInputs = [ autoPatchelfHook desktop-file-utils ];
 
-  installPhase = ''
-    runHook preInstall
+in appimageTools.wrapType2 rec {
+  inherit name src;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/standardnotes/default.nix:20]:
```suggestion
  inherit pname src;
```

File: pkgs/applications/editors/standardnotes/default.nix [hunk 4]
```diff
@@ -1,45 +1,38 @@
...
-  inherit (srcjson) version;
-
-  dontConfigure = true;
-
-  dontBuild = true;
+  src = fetchurl (srcjson.appimage.${stdenv.hostPlatform.system} or throwSystem);
 
-  nativeBuildInputs = [ makeWrapper dpkg desktop-file-utils ];
+  appimageContents = appimageTools.extract {
+    inherit name src;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/standardnotes/default.nix:14]:
```suggestion
    inherit pname src;
```

File: pkgs/applications/editors/standardnotes/default.nix [hunk 5]
```diff
@@ -1,45 +1,38 @@
...
-
-  throwSystem = throw "Unsupported system: ${stdenv.hostPlatform.system}";
-
-in
-
-stdenv.mkDerivation rec {
-
+  version = srcjson.version;
   pname = "standardnotes";
+  name = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/standardnotes/default.nix:8]:
```suggestion
```
all appimage tools now support pname, please use that instead

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
> Is it worth investigating why standardnotes with nixpkgs's electron has this issue, or filing upstream somewhere?

Yes, it is definitely worth it. Are we matching upstreams version? If not that would be my biggest bread crumb to start with.

File: pkgs/applications/editors/standardnotes/default.nix (hunk 3 above)
Comment (duxovni) [reply pkgs/applications/editors/standardnotes/default.nix:20]: And likewise here?
```suggestion
  inherit pname version src;
```

**Comment by duxovni** [comment]
Closing this PR since #247053 appears to be resolved.

**Pull request closed without merging**